pkg/network/usm/sharedlibraries/testutil/file_opener.go

```diff
@@ -0,0 +1,104 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package testutil
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// mutex protecting build process
+var mux sync.Mutex
+var cachedArtifactPath string
+
+func OpenFromAnotherProcess(t *testing.T, paths ...string) *exec.Cmd {
+	build(t)
+
+	cmd := exec.Command(cachedArtifactPath, paths...)
+	require.NoError(t, cmd.Start())
+
+	t.Cleanup(func() {
+		if cmd.Process == nil {
+			return
+		}
+		_ = cmd.Process.Kill()
+	})
+
+	return cmd
+}
+
+func build(t *testing.T) {
+	mux.Lock()
+	defer mux.Unlock()
+
+	if cachedArtifactPath != "" {
+		return
+	}
```

Comment:
you're implementing `sync.Once`, unless we have a good reason, I vote for moving into `sync.Once`